### PR TITLE
How to Encode examples

### PR DESCRIPTION
**Pull request by remiadon**
This PR introduces a set of examples for our end-users to quickly identify a design pattern that has already been used in an other workflow, and make proper use of BIDS-prov to encode their own workflow

TODO
 - [ ] add more examples and make sure they are ready for communication
 - [x] check examples provide the fields described in the [GDOC](https://docs.google.com/document/d/1vw3VNDof5cecv2PkFp7Lw_pNUTUo8-m8V4SIdtGJVKs/edit#)
 - [ ] provide a visual overview on the critical questions when using BIDS-prov. Suggestion : a decision tree, [like in scikit-learn](https://scikit-learn.org/stable/tutorial/machine_learning_map/index.html)

### CODE REVIEW

File: how-to-encode.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
+## How to encode my workflow given the BIDS-prov ontology ?
+
+Either you are a software developper, a researcher striving for reproducible science, or anyone working in the neuro-imaging field and willing to 
+use BIDS-prov, at some point you might be asking yourself the following question :
+
+> What is the right way to represent my workflow given in the philosophy of BIDS-prov ?
```
Comment (remiadon) [inline how-to-encode.md:6]: [RM] "given"

File: how-to-encode.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        "used": "niiri:entity_1"
+      }
+    ]
+    "prov:Entity": [
+      {
+        "@id": "niiri:entity_1",
+        "label": "Entity 1",
+        "prov:atLocation": "/root/file.xyz",
+        "generatedAt": "2019-10-10T10:00:00"
+      },
```
Comment (remiadon) [inline how-to-encode.md:85]: [ADD] closing bracket (end of list)

File: how-to-encode.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        "command": "launch.sh"
+      },
+      {
+        "@id": "niiri:activity_1",
+        "label": "Activity 1",
+        "command": "task_specific_executable.sh --arg 1",
+        "parameters": {
+            "arg1" : 1
+        },
+        "wasAssociatedWith": "RRID:SCR_007037",
```
Comment (remiadon) [inline how-to-encode.md:73]: isPartOf below

Comment (remiadon) [reply how-to-encode.md:73]: hum actually there is not such isPartOf relation

@cmaumet do you have any idea of how to encode this ? Should we remove this example ?

**Comment by remiadon** [comment]
Other ideas of examples 
- "Should I always declare provenance in a single file ?"  -> give directions so that "visualizing" the files separately still make sense. And explicitly state that NO, you can declare multiple files that share the same context
- "I want to track provenance for subject-level analysis, should I declare a single prov file for every subject ?"

**Comment by cmaumet** [comment]
@remiadon - really like the how-to-encode.md file! Could we include this info directly in the spec?

**Comment by remiadon** [comment]
@cmaumet Yes, I just needed a place to clearly think about this

Do you have more use cases in mind before we merge this into the spec ?

**Comment by cmaumet** [comment]
Hey @remiadon! Please reply to my email about this afternoon's meeting. This is getting very urgent now.

File: how-to-encode.md [hunk 4]
```diff
@@ -0,0 +1,168 @@
+# How to encode my workflow given the BIDS-prov ontology ?
+
+Either you are a software developper, a researcher striving for reproducible science, or anyone working in the neuro-imaging field and willing to 
+use BIDS-prov, at some point you might be asking yourself the following question :
+
+> What is the right way to represent my workflow in the philosophy of BIDS-prov ?
+
+This set of examples will give you an overview the the typical cases and how to apply BIDS-prov concepts !
```
Comment (remiadon) [inline how-to-encode.md:8]: remove the

File: how-to-encode.md [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+
+
+------------
+------------
+
+## I want to track provenance for subject-level analysis, should I declare a single prov file for every subject ?
+
+You can create a single prov file for every subject.
+
+Yey another option is to use globbing to group multiple files into the same entity, using globbing in the `generatedAt` field of this entity.
```
Comment (remiadon) [inline how-to-encode.md:68]: Yet

**Comment by remiadon** [comment]
> @remiadon - really like the how-to-encode.md file! Could we include this info directly in the spec?

@cmaumet done. There is a section called "6 How to encode your workflow with BIDS-prov" in suggestion mode

**Pull request closed without merging**